### PR TITLE
Closes #5 | Add tatoeba dataset loader

### LINKED ISSUE #5: Create dataset loader for Tatoeba
Dataloader name: `tatoeba/tatoeba.py`
DataCatalogue: http://seacrowd.github.io/seacrowd-catalogue/card.html?tatoeba

|     Dataset        | tatoeba  |
|-------------|---|
| Description | Tatoeba is a collection of sentences and translations collected by Tatoeba.org |
| Subsets | tatoeba.ind, tatoeba.jav, tatoeba.tgl, tatoeba.tha, tatoeba.vie |
| Languages | ind, vie, tgl, jav, tha |
| License | Other (other) |
| Homepage | https://tatoeba.org |
| HF URL | https://huggingface.co/datasets/xtreme/ |
| Paper URL | https://tatoeba.org/en/ |

### PR DESCRIPTION
**Pull request by ljvmiranda921**
Closes #5

Some notable things to point out from this dataset:
- Instead of implementing one loader per language, I created several configs per language in a single loader. TThis means that the configs would look like this: `tatoeba.ind_seacrowd_t2t`, `tatoeba.ind_source` and so on. When testing, we should pass a value to the `--subset_id` parameter like so:

```
python -m tests.test_seacrowd seacrwod/sea_datasets/tatoeba/tatoeba.py --subset_id tatoeba.ind
```

- The [`source_lang` field in the original HF link](https://huggingface.co/datasets/xtreme/viewer/tatoeba.tgl) seems incorrect?  I'm not familiar with this dataset so for my implementation I just copied over what's in the source. Let me know if I should "override" this with the actual language code.

### Checkbox
- [X] Confirm that this PR is linked to the dataset issue.
- [x] Create the dataloader script `seacrowd/sea_datasets/my_dataset/my_dataset.py` (please use only lowercase and underscore for dataset naming).
- [x] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_SEACROWD_VERSION` variables.
- [x] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [x] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `SEACrowdConfig` for the source schema and one for a seacrowd schema.
- [x] Confirm dataloader script works with `datasets.load_dataset` function.
- [x] Confirm that your dataloader script passes the test suite run with `python -m tests.test_seacrowd seacrowd/sea_datasets/<my_dataset>/<my_dataset>.py`.
- [ ] If my dataset is local, I have provided an output of the unit-tests in the PR (please copy paste). This is OPTIONAL for public datasets, as we can test these without access to the data files.

### CODE REVIEW

File: seacrowd/sea_datasets/tatoeba/tatoeba.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+        source_file = filepath[0]
+        target_file = filepath[1]
+        source_sentences = []
+        target_sentences = []
+        with open(source_file, encoding="utf-8") as f1:
+            for row in f1:
+                source_sentences.append(row)
+        with open(target_file, encoding="utf-8") as f2:
+            for row in f2:
+                target_sentences.append(row)
```
Comment (ryanignatius) [inline seacrowd/sea_datasets/tatoeba/tatoeba.py:114]: each row here contains `\n`,
I think we should strip it

**Comment by ryanignatius** [comment]
> The [source_lang field in the original HF link](https://huggingface.co/datasets/xtreme/viewer/tatoeba.tgl) seems incorrect? I'm not familiar with this dataset so for my implementation I just copied over what's in the source. Let me know if I should "override" this with the actual language code.

About this, I think it will be better to "override" this with the actual language code,
What do you think @SamuelCahyawijaya ?

File: seacrowd/sea_datasets/tatoeba/tatoeba.py (hunk 1 above)
Comment (ljvmiranda921) [reply seacrowd/sea_datasets/tatoeba/tatoeba.py]: Done! https://github.com/SEACrowd/seacrowd-datahub/pull/22/commits/0ee89229f3717e1d8ece034efc0cd9d6de1d04e9

**Comment by holylovenia** [comment]
@ryanignatius I agree with you. Could you please override it with the actual language codes? @ljvmiranda921

**Comment by ljvmiranda921** [comment]
Should be done: https://github.com/SEACrowd/seacrowd-datahub/pull/22/commits/aba9c65759fc67ac114cd8bcd1b6388a296211fc

File: seacrowd/sea_datasets/tatoeba/tatoeba.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+class TatoebaDatset(datasets.GeneratorBasedBuilder):
+    """Tatoeba subset for Indonesian, Vietnamese, Tagalog, Javanese, and Thai."""
+
+    SOURCE_VERSION = datasets.Version(_SOURCE_VERSION)
+    SEACROWD_VERSION = datasets.Version(_SEACROWD_VERSION)
+
+    SEACROWD_SCHEMA_NAME = "t2t"
+
+    dataset_names = sorted([f"tatoeba.{lang}" for lang in _LANGUAGES])
```
Comment (holylovenia) [inline seacrowd/sea_datasets/tatoeba/tatoeba.py:52]: Instead of using `.` as the delimiter, could you please change it to `_`?

Comment (ljvmiranda921) [reply seacrowd/sea_datasets/tatoeba/tatoeba.py]: Sure! https://github.com/SEACrowd/seacrowd-datahub/pull/22/commits/9317deaab0daa76ced2757c2000aa79678f7c287

**Comment by ljvmiranda921** [comment]
> could you also please provide tatoeba_source and tatoeba_seacrowd_t2t schemas where they load all data?

Hi @holylovenia ! I implemented it here: https://github.com/SEACrowd/seacrowd-datahub/pull/22/commits/a44be2b0edea85af8a7423ea9a4427c08f17d0e6

The default config now is `tatoeba_source` (loads all subsets from every SEA language using the `source` schema).

**Review by holylovenia — APPROVED** [review]
Thank you so much for the changes, @ljvmiranda921! Looks perfect to me. 👍  Let's wait for @ryanignatius's review then we can merge it.

**Review by ryanignatius — APPROVED** [review]
it looks good to me too,
thank you for the help @ljvmiranda921

**Pull request merged**